signer/src/stacks/api.rs

```diff
@@ -547,14 +572,57 @@ impl StacksClient {
 }
 
 impl StacksInteract for StacksClient {
-    async fn get_block(&self, block_id: StacksBlockId) -> Result<NakamotoBlock, Error> {
-        self.get_block(block_id).await
+    async fn get_account(&mut self, address: &StacksAddress) -> Result<AccountInfo, Error> {
+        let result = StacksClient::get_account(self, address).await;
+        match result {
+            Err(Error::StacksNodeRequest(_)) | Err(Error::StacksNodeResponse(_)) => {
+                self.next_endpoint();
+            }
+            _ => (),
+        }
+        result
     }
-    async fn get_tenure(&self, block_id: StacksBlockId) -> Result<Vec<NakamotoBlock>, Error> {
-        self.get_tenure(block_id).await
+
+    async fn submit_tx(&mut self, tx: &StacksTransaction) -> Result<SubmitTxResponse, Error> {
+        let result = StacksClient::submit_tx(self, tx).await;
+        match result {
+            Err(Error::StacksNodeRequest(_)) | Err(Error::StacksNodeResponse(_)) => {
+                self.next_endpoint();
+            }
+            _ => (),
+        }
+        result
     }
-    async fn get_tenure_info(&self) -> Result<RPCGetTenureInfo, Error> {
-        self.get_tenure_info().await
+
+    async fn get_block(&mut self, block_id: StacksBlockId) -> Result<NakamotoBlock, Error> {
+        let result = StacksClient::get_block(self, block_id).await;
+        match result {
+            Err(Error::StacksNodeRequest(_)) | Err(Error::StacksNodeResponse(_)) => {
+                self.next_endpoint();
+            }
+            _ => (),
+        }
+        result
+    }
+    async fn get_tenure(&mut self, block_id: StacksBlockId) -> Result<Vec<NakamotoBlock>, Error> {
+        let result = StacksClient::get_tenure(self, block_id).await;
+        match result {
+            Err(Error::StacksNodeRequest(_)) | Err(Error::StacksNodeResponse(_)) => {
+                self.next_endpoint();
+            }
+            _ => (),
+        }
+        result
+    }
+    async fn get_tenure_info(&mut self) -> Result<RPCGetTenureInfo, Error> {
+        let result = StacksClient::get_tenure_info(self).await;
+        match result {
+            Err(Error::StacksNodeRequest(_)) | Err(Error::StacksNodeResponse(_)) => {
+                self.next_endpoint();
+            }
+            _ => (),
+        }
+        result
```

Comment:
I think it would better to just create a function for this logic. I believe most of these functions can lean on the same logic here.

Reply:
Done